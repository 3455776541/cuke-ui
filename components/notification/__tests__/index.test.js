import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Notification from "../index";

describe("<Notification/>", () => {
  it("should render a <Notification/> components", () => {
    const wrapper = render(
      <div>
        <Notification title="哈哈" type="success" />
        <Notification title="哈哈" type="error" />
        <Notification title="哈哈" type="info" />
        <Notification title="哈哈" type="warning" />
        <Notification title="哈哈" type="loading" />
        <Notification title="哈哈" type="open" />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should 2s ago emit callback", () => {
    const wrapper = shallow(
      <Notification
        title="哈哈"
        type="success"
        duration={2}
        onClose={() => wrapper.setProps({ title: "回调" })}
      />
    );

    setTimeout(() => {
      assert(wrapper.props().title === "回调");
    }, 2000);
  });

  it("should trigger onClose event", () => {
    const onClose = jest.fn();
    const wrapper = shallow(
      <Notification
        title="哈哈"
        type="success"
        duration={2}
        onClose={onClose}
      />
    );
    wrapper.find(".cuke-notification-close-btn").simulate("click");
    expect(onClose).toHaveBeenCalled();
  });

  it("should render destroy reference", () => {
    const notification = Notification.success();
    assert(notification.destroy && notification.destroy instanceof Function);
  });

  it("should cannot find <Notification/> when destroy", () => {
    const message = Notification.success();
    message.destroy();
    setTimeout(() => {
      assert(document.querySelector(".cuke-notification-success").length === 0);
    }, 10);
  });

  it("should render <Notification/> when call notification success static method", () => {
    Notification.success();
    setTimeout(() => {
      assert(document.querySelector(".cuke-notification-success").length === 1);
    }, 10);
  });
  it("should render <Notification/> when call notification error static method", () => {
    Notification.error();
    setTimeout(() => {
      assert(document.querySelector(".cuke-notification-error").length === 1);
    }, 10);
  });
  it("should render <Notification/> when call notification info static method", () => {
    Notification.info();
    setTimeout(() => {
      assert(document.querySelector(".cuke-notification-info").length === 1);
    }, 10);
  });
  it("should render <Notification/> when call notification warning static method", () => {
    Notification.warning();
    setTimeout(() => {
      assert(document.querySelector(".cuke-notification-warning").length === 1);
    }, 10);
  });
  it("should render <Notification/> when call notification loading static method", () => {
    Notification.loading();
    setTimeout(() => {
      assert(document.querySelector(".cuke-notification-loading").length === 1);
    }, 10);
  });

  it("should 2s ago emit onClick", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Notification
        title="哈哈"
        type="success"
        duration={2}
        onClick={onClick}
      />
    );
    wrapper.find(".cuke-notification").simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
  it("should disabled scroll when did update", () => {
    const wrapper = shallow(<Notification title="关闭回调" />);
    wrapper.update();
    assert(document.body.style.overflow === "");
  });
});
