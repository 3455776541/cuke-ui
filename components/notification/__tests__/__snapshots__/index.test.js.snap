// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification/> should render a <Notification/> components 1`] = `
<div>
  <div
    class="cuke-notification cuke-notification-open cuke-notification-open-top-right"
    style="top:20px"
  >
    <div
      class="cuke-notification-close-btn"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-icon cuke-notification-success"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
        />
        <polyline
          points="22 4 12 14.01 9 11.01"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-title-custom"
    >
      <div
        class="cuke-notification-title"
      >
        <span>
          哈哈
        </span>
      </div>
      <div
        class="cuke-notification-message"
      />
    </div>
  </div>
  <div
    class="cuke-notification cuke-notification-open cuke-notification-open-top-right"
    style="top:20px"
  >
    <div
      class="cuke-notification-close-btn"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-icon cuke-notification-error"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="15"
          x2="9"
          y1="9"
          y2="15"
        />
        <line
          x1="9"
          x2="15"
          y1="9"
          y2="15"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-title-custom"
    >
      <div
        class="cuke-notification-title"
      >
        <span>
          哈哈
        </span>
      </div>
      <div
        class="cuke-notification-message"
      />
    </div>
  </div>
  <div
    class="cuke-notification cuke-notification-open cuke-notification-open-top-right"
    style="top:20px"
  >
    <div
      class="cuke-notification-close-btn"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-icon cuke-notification-info"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <polygon
          points="11 5 6 9 2 9 2 15 6 15 11 19 11 5"
        />
        <path
          d="M19.07 4.93a10 10 0 0 1 0 14.14M15.54 8.46a5 5 0 0 1 0 7.07"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-title-custom"
    >
      <div
        class="cuke-notification-title"
      >
        <span>
          哈哈
        </span>
      </div>
      <div
        class="cuke-notification-message"
      />
    </div>
  </div>
  <div
    class="cuke-notification cuke-notification-open cuke-notification-open-top-right"
    style="top:20px"
  >
    <div
      class="cuke-notification-close-btn"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-icon cuke-notification-warning"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="8"
          y2="12"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="16"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-title-custom"
    >
      <div
        class="cuke-notification-title"
      >
        <span>
          哈哈
        </span>
      </div>
      <div
        class="cuke-notification-message"
      />
    </div>
  </div>
  <div
    class="cuke-notification cuke-notification-open cuke-notification-open-top-right"
    style="top:20px"
  >
    <div
      class="cuke-notification-close-btn"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-icon cuke-notification-loading"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="6"
        />
        <line
          x1="12"
          x2="12"
          y1="18"
          y2="22"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="4.93"
          y2="7.76"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="16.24"
          y2="19.07"
        />
        <line
          x1="2"
          x2="6"
          y1="12"
          y2="12"
        />
        <line
          x1="18"
          x2="22"
          y1="12"
          y2="12"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="19.07"
          y2="16.24"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="7.76"
          y2="4.93"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-title-custom"
    >
      <div
        class="cuke-notification-title"
      >
        <span>
          哈哈
        </span>
      </div>
      <div
        class="cuke-notification-message"
      />
    </div>
  </div>
  <div
    class="cuke-notification cuke-notification-open cuke-notification-open-top-right"
    style="top:20px"
  >
    <div
      class="cuke-notification-close-btn"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </div>
    <div
      class="cuke-notification-icon cuke-notification-open"
    />
    <div
      class="cuke-notification-title-custom"
    >
      <div
        class="cuke-notification-title"
      >
        <span>
          哈哈
        </span>
      </div>
      <div
        class="cuke-notification-message"
      />
    </div>
  </div>
</div>
`;
